### PR TITLE
FE-928: Re-style wrapper global components

### PR DESCRIPTION
**Pull request by shrirupa**
FE-928: Re-style wrapper global components

### What Changed

**There are some changes to CardTitle original css, switched the color to #55555a or color('gray',2) or sparkpost brand color, which seems to be the one used in mocks and also some adjustments to font size to accommodate different font sizes**
link to mocks-
https://sparkpost.invisionapp.com/share/BMU7BR332NK#/screens/386841857
https://sparkpost.invisionapp.com/share/XDSMDD078FT#/screens

wrapper global components for Hibana updated:
```
<ButtonWrapper/>
<Callout/>
<Card/>
<Empty/>
```
- Updated component styles for Hibana

### How To Test
 - `<Card>`
    - /onboarding/recipient-validation
Screen Shot 2020-03-31 at 1 14 54 PM
    

- /dashboard

Screen Shot 2020-03-31 at 11 42 33 AM

Screen Shot 2020-03-31 at 1 19 06 PM

- `<Empty />`
Screen Shot 2020-03-31 at 5 07 05 PM

### To Do
- [ ] Address feedback

### CODE REVIEW

File: src/components/card/Card.js [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
+    <Box display="inline-block" fontSize="500">
+      {children}
+    </Box>
+  );
+};
+
+export const CardTitle = ({ children }) => {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  if (!isHibanaEnabled) return <h2 className={styles.CardTitle}>{children}</h2>;
```
Comment (bkemper) [inline src/components/card/Card.js:36]: Should we make any changes to the current component?

File: src/components/buttonWrapper/ButtonWrapper.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
-const ButtonWrapper = ({ children }) => (
-  <div className={styles.ButtonWrapper}>
-    {children}
-  </div>
-);
+export const ButtonWrapper = ({ children }) => {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  if (!isHibanaEnabled) return <div className={styles.ButtonWrapper}>{children}</div>;
+  return <Box marginTop="500">{children}</Box>;
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/components/buttonWrapper/ButtonWrapper.js:10]: This is a token, actually - passing in these values as strings is how you access the tokens from the theme provider

[the remark this answers is not in the document]
Comment (shrirupa) [reply src/components/buttonWrapper/ButtonWrapper.js:10]: Tokens, I think 🤷‍♀ .

File: src/components/buttonWrapper/ButtonWrapper.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+};
 
 export default ButtonWrapper;
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/components/buttonWrapper/ButtonWrapper.js:13]: I can add a snapshot test.

File: src/components/card/Card.js (hunk 1 above)
Comment (shrirupa) [reply src/components/card/Card.js:36]: I wasn't completely sure of this. but did this for RVBundlePage. 
We are using <CardTitle> in only two places in the app (**RVBundlePage** and **IpForms**, FeatureStepList is disabled so not in use anymore).

File: src/components/callout/Callout.js [hunk 4]
```diff
@@ -1,20 +1,37 @@
...
+        <div>
+          <h3 className={styles.Title}>{title}</h3>
+          {children && <p className={styles.Content}>{children}</p>}
+        </div>
+      </div>
+    );
+  return (
+    <Box height={height} display="flex" justifyContent="center" alignItems="center">
+      <Box>
+        <Text as="h3" lineHeight="2">
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/components/callout/Callout.js:22]: Agreed with Brian here!

File: src/components/callout/Callout.js [hunk 5]
```diff
@@ -1,20 +1,37 @@
...
+      <div className={styles.Callout} style={{ height }}>
+        <div>
+          <h3 className={styles.Title}>{title}</h3>
+          {children && <p className={styles.Content}>{children}</p>}
+        </div>
+      </div>
+    );
+  return (
+    <Box height={height} display="flex" justifyContent="center" alignItems="center">
+      <Box>
```
Comment (nicklemmon) [inline src/components/callout/Callout.js:21]: Is this extra `<Box/>` doing anything?

File: src/components/callout/Callout.test.js [hunk 6]
```diff
@@ -1,20 +1,27 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { mount } from 'enzyme';
```
Comment (nicklemmon) [inline src/components/callout/Callout.test.js:2]: 👏

File: src/components/callout/Callout.test.js [hunk 7]
```diff
@@ -1,20 +1,27 @@
...
-      </Callout>
+    const wrapper = mount(
+      <TestApp>
+        <Callout title="Example">Here is the deal...</Callout>
+      </TestApp>,
     );
 
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('h3').text()).toEqual('Example');
+    expect(wrapper.find('p').text()).toEqual('Here is the deal...');
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/components/callout/Callout.test.js:25]: Agreed!

File: src/components/card/Card.js (hunk 1 above)
Comment (nicklemmon) [reply src/components/card/Card.js:36]: Yeah the original semantics were more appropriate, actually - `<div role="heading" aria-level="3"/>` is the equivalent of an `<h3>` not an `<h2>`, but I think in general we don't want to change the original components.

File: src/components/card/Card.module.scss [hunk 8]
```diff
@@ -11,9 +11,7 @@
 
 .CardTitle {
   display: block;
-  color: color(gray, 3);
```
Comment (nicklemmon) [inline src/components/card/Card.module.scss:14]: Are these styles not needed for the original component?

File: src/components/empty/Empty.js [hunk 9]
```diff
@@ -1,24 +1,36 @@
...
       <Panel sectioned title={title}>
         <h6 className={styles.Center}>{message}</h6>
       </Panel>
     );
   }
+  return (
+    <Panel sectioned title={title}>
+      <Box textAlign="center" color="gray.400">
+        <Block size={28} />
+        <Text as="h3" color="gray.400">
```
Comment (nicklemmon) [inline src/components/empty/Empty.js:23]: 👍 This is a better choice semantically. Nice.

File: src/components/card/Card.js [hunk 10]
```diff
@@ -1,15 +1,43 @@
...
-    {children}
-  </div>
-);
+export const Card = ({ children, textAlign }) => {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  if (!isHibanaEnabled)
+    return <div className={classNames(styles.CardContainer, styles[textAlign])}>{children}</div>;
+  return (
+    <Box border="400" padding="400" textAlign={textAlign}>
```
Comment (nicklemmon) [inline src/components/card/Card.js:13]: I think the Hibana color styles need to be applied to these components - right now those colors are being inherited from global styles - not ideal!

File: src/components/empty/Empty.js [hunk 11]
```diff
@@ -1,24 +1,36 @@
...
+  if (!isHibanaEnabled) {
     return (
       <Panel sectioned title={title}>
         <h6 className={styles.Center}>{message}</h6>
       </Panel>
     );
   }
+  return (
+    <Panel sectioned title={title}>
+      <Box textAlign="center" color="gray.400">
```
Comment (nicklemmon) [inline src/components/empty/Empty.js:21]: This color does not meet color contrast requirements! Even if the mocks indicate a lighter color, we should use the lightest possible value that still passes. You can use this to check the values, or install the Axe Chrome extension:
https://color.review/

Screen Shot 2020-04-01 at 12 34 50 PM

File: src/components/empty/Empty.js (hunk 9 above)
Comment (nicklemmon) [reply src/components/empty/Empty.js:23]: Same as above - the color is too light

File: src/pages/onboarding/RVBundlePage.js [hunk 12]
```diff
@@ -55,33 +61,35 @@ const RVBundlePage = () => {
...
-              <p>
-                With our monthly pay-as-you-go plan using tiered pricing, the more you validate, the
-                less you pay per message.
-              </p>
-              <p>
-                <UnstyledLink onClick={() => setModalOpen(true)} className={styles.link}>
-                  See Pricing
-                </UnstyledLink>
-              </p>
+              </CardTitle>
```
Comment (nicklemmon) [inline src/pages/onboarding/RVBundlePage.js:67]: If this one _needs_ to be an `<h2>` then you could make a `level` prop that defaults to `"3"`?

File: src/components/empty/tests/Empty.test.js [hunk 13]
```diff
@@ -1,10 +1,16 @@
...
   it('should render', () => {
-    const wrapper = shallow(<Empty title={'Empty Title'} message={'nothing here to see'} />);
-    expect(wrapper).toMatchSnapshot();
+    const wrapper = mount(
+      <TestApp>
+        <Empty title={'Empty Title'} message={'nothing here to see'} />
+      </TestApp>,
+    );
+    expect(wrapper.find('Panel')).toHaveLength(1);
+    expect(wrapper.find('h6')).toHaveLength(1);
```
Comment (nicklemmon) [inline src/components/empty/tests/Empty.test.js:14]: Might be better to use `.toHaveTextContent()` again

File: src/components/buttonWrapper/ButtonWrapper.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bkemper) [reply src/components/buttonWrapper/ButtonWrapper.js:10]: My question wasn't very good.  Take two...

Do we want to define style-system props with static values verse explicitly with the token?

```
<Box marginTop="500">
```

vs 

```
<Box marginTop={tokens.spacing_500}>
```

[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/components/buttonWrapper/ButtonWrapper.js:10]: The static values are actually some syntactic sugar that `styled-system` provides - it directly hooks in to the theme provider so we don't have to import the tokens in to every component, reducing a lot of boilerplate.

Not only that, but it's a form of restriction that prevents someone from passing in the wrong token. If you pass in tokens explicitly, technically this works:
```
<Box marginTop={tokens.borderRadius_200}/>
```

Furthermore, if the token name changes, it will break - using `"200"` keeps those two things separate!

[the remark this answers is not in the document]
Comment (bkemper) [reply src/components/buttonWrapper/ButtonWrapper.js:10]: Cool, can you summarize this decision and share with everyone in public guild channel?

File: src/components/card/Card.module.scss (hunk 8 above)
Comment (shrirupa) [reply src/components/card/Card.module.scss:14]: Since I had changed the `<CardTitle>` component so I updated these to make sure they look same visually. Reverting it.

File: src/components/card/Card.js [hunk 14]
```diff
@@ -1,15 +1,63 @@
...
+const HibanaCardTitle = ({ children }) => {
+  return (
+    <Text as="h3" mb="100" color="gray.900">
+      {children}
+    </Text>
+  );
+};
+
+export const Card = ({ children, textAlign }) => {
+  return useHibanaToggle(OGCard, HibanaCard)({ children, textAlign });
```
Comment (nicklemmon) [inline src/components/card/Card.js:54]: It's probably simpler to just pass in `(props)` instead of doing the destructuring an extra time. Just my two cents - not a critical problem or anything.

File: src/components/buttonWrapper/ButtonWrapper.js [hunk 15]
```diff
@@ -1,10 +1,17 @@
...
+};
+
+const HibanaButtonWrapper = ({ children }) => {
+  return <Box marginTop="500">{children}</Box>;
+};
+export const ButtonWrapper = ({ children }) => {
+  return useHibanaToggle(OGButtonWrapper, HibanaButtonWrapper)({ children });
+};
 
 export default ButtonWrapper;
```
Comment (nicklemmon) [inline src/components/buttonWrapper/ButtonWrapper.js:17]: Do both need to be exported in this way? I would think you could just stick with the default export.

File: src/components/callout/Callout.js [hunk 16]
```diff
@@ -1,8 +1,25 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './Callout.module.scss';
+import useHibanaToggle from 'src/components/matchbox/useHibanaToggle';
+import { Box, Text } from 'src/components/matchbox';
 
-const Callout = ({ children, height = '220px', title }) => (
+const HibanaCallout = ({ children, height = '220px', title }) => (
+  <Box height={height} display="flex" justifyContent="center" alignItems="center">
+    <Box>
```
Comment (nicklemmon) [inline src/components/callout/Callout.js:9]: Is the extra `<Box/>` necessary? Maybe the flexbox styles aren't necessary? I don't know, just seems a little unusual! I could be wrong.

File: src/components/callout/Callout.test.js [hunk 17]
```diff
@@ -1,20 +1,27 @@
...
 
   it('renders title and details', () => {
-    const wrapper = shallow(
-      <Callout title="Example">
-        Here is the deal...
-      </Callout>
+    const wrapper = mount(
+      <TestApp>
+        <Callout title="Example">Here is the deal...</Callout>
+      </TestApp>,
```
Comment (nicklemmon) [inline src/components/callout/Callout.test.js:21]: 👏 Death to shallow mounting! Nice work.

File: src/components/card/Card.js [hunk 18]
```diff
@@ -1,15 +1,63 @@
...
+const OGCardTitle = ({ children, level = '3' }) => (
+  <div
+    className={classNames(
+      styles.CardTitle,
+      level === '2' && styles.level2,
+      level === '3' && styles.level3,
+    )}
+    role="heading"
+    aria-level={level}
+  >
```
Comment (nicklemmon) [inline src/components/card/Card.js:24]: 👏

File: src/components/card/Card.js [hunk 19]
```diff
@@ -1,15 +1,63 @@
...
+  return (
+    <Box display="inline-block" fontSize="500" color="gray.900">
+      {children}
+    </Box>
+  );
+};
+
+const HibanaCardTitle = ({ children }) => {
+  return (
+    <Text as="h3" mb="100" color="gray.900">
```
Comment (nicklemmon) [inline src/components/card/Card.js:47]: Probably worth supporting `level` here as well - might even mean doing `<Text as="div" role="heading" level={level}>`

File: src/components/matchbox/useHibanaToggle.js [hunk 20]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { useHibana } from 'src/context/HibanaContext';
+
+export default function useHibanaToggle(OGComponent, HibanaComponent) {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  return props => (isHibanaEnabled ? <HibanaComponent {...props} /> : <OGComponent {...props} />);
+}
```
Comment (nicklemmon) [inline src/components/matchbox/useHibanaToggle.js:8]: Code looks good - I would think this should either live with Hibana code or in `src/hooks`? This doesn't really have much to do with Matchbox.

File: src/pages/onboarding/RVBundlePage.js [hunk 21]
```diff
@@ -55,33 +61,35 @@ const RVBundlePage = () => {
         <Panel.Section>
           <div className={styles.grid}>
             <Card>
-              <h2>
-                <CheckCircle className={styles.icon} />
+              <CardTitle level="2">
```
Comment (nicklemmon) [inline src/pages/onboarding/RVBundlePage.js:64]: 👍 Nice and clean.

File: src/components/callout/Callout.js [hunk 22]
```diff
@@ -1,8 +1,25 @@
...
+      {children && (
+        <Text as="p" color="gray.800">
+          {children}
+        </Text>
+      )}
+    </Box>
+  </Box>
+);
+
+const OGCallout = ({ children, height = '220px', title }) => (
```
Comment (nicklemmon) [inline src/components/callout/Callout.js:22]: Screen Shot 2020-04-03 at 9 47 44 AM

It looks like `height` isn't making it for some reason?

Comment (shrirupa) [reply src/components/callout/Callout.js:22]: 😱

File: src/components/card/Card.js (hunk 19 above)
Comment (shrirupa) [reply src/components/card/Card.js:47]: looks like aria-level does't work for `<Text>`
